src/main/java/net/dv8tion/jda/internal/handle/SocketHandler.java

```diff
@@ -33,10 +35,13 @@ public final synchronized void handle(long responseTotal, DataObject o)
     {
         this.allContent = o;
         this.responseNumber = responseTotal;
-        final Long guildId = handleInternally(o.getObject("d"));
+        final DataObject data = o.getObject("d");
+        if (getJDA().isEventPassthrough()) CURRENT_EVENT.set(data);
```

Comment:
Not sure I agree with this.
Providing access to the `t` as well would allow users to look at the documentation to know what event the data is for when looking at the Discord Documentation.

I'd rather we just send down the `o` here than pulling out the `d`

Reply:
It would also match the format of what we send in `RawGatewayEvent`

Reply:
I don't think it is worth doing that, since if you want to handle that level you should be using `RawGatewayEvent`. The point of the passthrough is to use it on specific events, where the type is already known.